### ISSUE TITLE
ordering seals via Aeon

### ISSUE THREAD

**Issue by holfordm** [bodleian/charters#56]
relates to #47 

A large number of seals from the charter collection were cut off and are now stored separately from the charters. E.g. https://charters-qa.bodleian.ox.ac.uk/catalog/charter_105091 MS. Ch. Suffolk 733 is the charter, MS. Ch. Suffolk 733* is the seal.

This potentially complicates the ordering process. A request at the charter_105091 page could trigger requests for both the charter and the seal, which might be easiest but would be inefficient if the reader didn't want to see the seal. Or could there be different buttons? - request charter, request seal, request charter and seal? 

To complicate things, one charter may have several seals: https://charters-qa.bodleian.ox.ac.uk/catalog/charter_105068

**Comment by Slange-Mhath** [comment]
Sorry for the delay, I am currently working on this and it seems a bit trickier than I previously though. Thanks to @irv I managed to get the desired functionality frontend wise: If there are detached seals it will now render a Request charter and request seal button. If not it will display request item. 
Trouble is behind the scenes nothing changes. So the request charters and the request seal buttons are currently doing the same things. For that to change I guess we'd need to change the values in the hiddern input forms which is whats happening behind the scenes if a button is clicked. It takes the solr values c from the **charter record** . To achieve the render of the additional button we use a bit of a hacky workaround currently, which is to check if the charters record has something in ms_shelfmarks_sm which is different to the title -> as it makes sense to assume that this is a seal then. This way we don't need to request the seal childDocument which does give me a major headache to do and would probably slow down the site quite heavily. 

If I am not completely wrong, the seal childDocs don't provide these needed input values anyway so we'd need to get them from the charters record (parent so to say). 

My idea would now be to just use the values _ms_requestitemid_sni, lang_sm, ms_availability_smni, ms_barcode_smni, ms_repository_s_ from the charters record and replace the title with the ms_shelfmarks_sm - but I am not sure if that would work as I don't know how it is set in aeon. Also that would cause the problem that there are multiple values for example: _MS. Ch. Berks. 20 b_, _MS. Ch. Berks. 20b*_ so I am not sure what to do in that case and which value to take. Also what if more than one seal is attached to a record? 

This is mainly me thinking out loud for documentation reasons but I'd also appreciate any thoughts on it.

**Comment by holfordm** [comment]
@Slange-Mhath and @irv thanks for this - as you've seen it's more complicated than I had realized both because of some charters having several seals and in terms of what needs to happen in the library when these items are fetched. As discussed earlier what I now suggest is this:

### charters with detached seals 
these have two request buttons (1) "request charter" (2) "how to request seals". (2) produces a pop up as https://medieval.bodleian.ox.ac.uk/catalog/manuscript_1167 "See the [Availability](https://medieval.bodleian.ox.ac.uk/catalog/manuscript_1167#availability) section of this record for information on viewing the item in a reading room." The availability section of relevant records will be changed to read `The seal(s) from this charter have been detached and are now stored separately. To request consultation please email <email>specialcollections.enquiries@bodleian.ox.ac.uk</email>.`. The `availability` has a `status="detachedSeal"` to make identifying these items straightforward.

### detached seals without associated charters
these have only one request button "how to request seals" with pop up as above and availability `To request consultation of this item please email <email>specialcollections.enquiries@bodleian.ox.ac.uk</email>`. The `availability` element has `status="sealOnly"` to make identifying these items straightforward.

### charters with attached seals
these just have the usual "request this item" button, no changes needed.

If this sounds sensible? I can make a start on implementing the TEI changes tomorrow.

**Comment by Slange-Mhath** [comment]
Hey Matthew, thanks for this. 

If I remember correctly, the difficulty was that a seal is not a similar entity as a charter so to say. A seal is a childDocument of the charters Document if that makes sense. I need to get my head around how that works for detached seals without a charter though ... 

We are currently trying to avoid to request all the childDocuments (seals) with every user query, which would probably have an impact on performance. 

Therefore we thought that we could identify detached seals via their charters shelfmarks. 

So for example: MS. Ch. Cambs 1 has the ms_shelfmarks_sm "MS. Ch. Cambs. 1", "MS. Ch. Cambs. 1*"

While the first one is the one for the charter, the second one must be for a seal and thus we can assume it has a seal (all credit to Andy). So we can simply check if there is a value which is different from the title in ms_shelfmarks_sm and render the additional button. 

However: 
I noticed that this does not work for /charter_800055 as it seems to not have another value in ms_shelfmarks_sm but displays on the show page that it has a detached seal. Any idea why that is? 

Also we wouldn't benefit from the availablity status I guess and it will make it quite tricky to identify the three cases: 

- Attached seal: 
Thats an easy one I guess, we check if ms_seals_b is yes and the logic described above returns false 

- Detached seal:
Basically the logic above (if we can figure out why its sometimes not working)
 
- Detached seal without charter:
That I am not sure, would you be able to point me to an example? 

I am not sure if I missed something or got something wrong? @irv

**Comment by irv** [comment]
I think for `tei-qa/MSS_Ch_Berks/MS_Ch_Berks_102.xml`

```xml
<authDesc>
  <seal>
    <p>(detached, kept separately)</p>
    <decoNote type="shield">Armorial - <persName><forename>Nicholas</forename> de <surname>Lychefeld</surname></persName></decoNote>
  </seal>
  <idno type="shelfmark">MS. Ch. Berks. 102*</idno>
</authDesc>
```

the `<idno>` should be in the `<seal>`, not a sibling of it. which is why it doesn't show up with it's own detatched seal identifier

**Comment by Slange-Mhath** [comment]
@holfordm Can you point me to an example for a detached seal which has no charter?

**Comment by holfordm** [comment]
https://charters.bodleian.ox.ac.uk/catalog/charter_200082

(anything with a * in the main shelfmark field should be a detached seal with no charter.) cf. https://github.com/bodleian/charters/commit/ec9d6013bb7bc7987b1bbca01e1c4baec8be4127  https://github.com/bodleian/charters/commit/40d80aa89166b80e3f7d0d627c3fec2b7b63c40a 
https://github.com/bodleian/charters/commit/0f35ef7abb9356d794f11207dde0057e8312fe4d

**Comment by holfordm** [comment]
@irv I think I have now updated the `availability` fields in the relevant records, so the ordering system currently live on QA can be set up on production when you have a chance. 
In case it helps to simplify the code, there are two different relevant `status` attributes for `availability`, `status=seal` for items which are only a seal (e.g. https://charters-qa.bodleian.ox.ac.uk/catalog/charter_109969) `status=detachedSeal` for charters with detached seal(s) (e.g. https://charters-qa.bodleian.ox.ac.uk/catalog/charter_120062)
